t/pipe.t

```diff
@@ -0,0 +1,1439 @@
+            local ok, err = proc:shutdown('stderr')
+            if not ok then
+                ngx.say(err)
+            end
+
+            local data = ("1234"):rep(2048)
+            local total = 0
+            local step = #data
+            -- make writers blocked later
+            while true do
+                local data, err = proc:write(data)
+                if not data then
+                    ngx.say(err)
+                    break
+                end
+
+                total = total + step
+                if total > 64 * step then
+                    break
+                end
+            end
+
+            ngx.thread.spawn(function() proc:write(data) end)
+            local ok, err = proc:shutdown('stdin')
+            if not ok then
+                ngx.say(err)
+            end
+        }
+    }
+--- response_body
+pipe busy reading
+pipe busy reading
+timeout
+pipe busy writing
+--- no_error_log
+[error]
+--- grep_error_log eval
+qr/lua pipe \w+ yielding/
+--- grep_error_log_out
+lua pipe read yielding
+lua pipe read yielding
+lua pipe write yielding
+lua pipe write yielding
+
+
+
+=== TEST 25: shutdown when merge_stderr is true
+--- config
+    location = /t {
+        content_by_lua_block {
+            local ngx_pipe = require "ngx.pipe"
+            local args = {"sh", "-c", "echo 'he\nllo' && >&2 echo 'wo\nrld'"}
+            local opts = {merge_stderr = true}
+            local function shutdown(proc, direction)
+                local ok, err = proc:shutdown(direction)
+                if not ok then
+                    ngx.say("shutdown: ", err)
+                end
+            end
+
+            local function read_proc(proc)
+                local ok, err = proc:stdout_read_line()
+                if not ok then
+                    ngx.say("stdout: ", err)
+                end
+            end
+
+            ngx.say("shutdown stdout after read")
+            local proc = ngx_pipe.spawn(args, opts)
+            proc:stdout_read_line()
+            shutdown(proc, 'stdout')
+            read_proc(proc)
+
+            ngx.say("shutdown stdout before read")
+            local proc = ngx_pipe.spawn(args, opts)
+            shutdown(proc, 'stdout')
+            read_proc(proc)
+
+            ngx.say("shutdown stderr")
+            local proc = ngx_pipe.spawn(args, opts)
+            shutdown(proc, 'stderr')
+        }
+    }
+--- response_body
+shutdown stdout after read
+stdout: closed
+shutdown stdout before read
+stdout: closed
+shutdown stderr
+shutdown: merged to stdout
+
+
+
+=== TEST 26: interrupt signals should not break ngx.pipe IO
+github issue openresty/resty-cli#35
+--- config
+    location = /t {
+        content_by_lua_block {
+            local ngx_pipe = require "ngx.pipe"
+            local filename = "$TEST_NGINX_HTML_DIR/testfile"
+            local f = io.open(filename, 'w')
+            f:write("testfile")
+            f:close()
```

Comment:
Is the intent here to reproduce the behavior observed in the linked GitHub issue via this `io.open` operation? Or something else?

Reply:
The `io.open` operation writes the `cpanfile` in that issue. The content in the file doesn't matter.